src/hooks/useCreateMap.ts

```diff
@@ -211,10 +215,10 @@ export const useCreateMap = () => {
     [createMapDoc, createObservationsDocs],
   )
 
-  function updateProgress() {
+  function updateProgress(uploads: UploadsList) {
```

Comment:
in order to avoid race conditions, I am passing the object everytime instead of just using the `UploadsRef`